app/components/UI/FiatOrders/orderProcessor/moonpay.js

```diff
@@ -0,0 +1,288 @@
+ * @property {string} state The customer's state, if the customer is from the USA. Returned as a two-letter code.
+ * @property {array} stages An array of four objects, each representing one of the four stages of the purchase process. The attributes of each stage are described below.
+ * @property {string} stage[stage] Possible values are stage_one_ordering, stage_two_verification, stage_three_processing, stage_four_delivery.
+ * @property {string} stage[status] Possible values are not_started, in_progress, success, failed.
+ * @property {string} stage[failureReason] Possible values are card_not_supported, daily_purchase_limit_exceeded, payment_authorization_declined, timeout_3d_secure, timeout_bank_transfer, timeout_kyc_verification, timeout_card_verification, rejected_kyc, rejected_card, rejected_other, cancelled, refund, failed_testnet_withdrawal, error. It can also be null.
+ * @property {array} stage[actions] Sometimes, the customer is required to take an action or actions to further the purchase process, usually by submitting information at a provided URL. For each action, we pass an object with a type and a url. Possible types are complete_bank_transfer, retry_kyc, verify_card_by_code, verify_card_by_file. If no actions are required, this returns an empty array.
+ */
+
+/**
+ * https://www.moonpay.com/dashboard/api_reference/client_side_api#transaction_object
+ * @enum {string}
+ */
+const MOONPAY_TRANSACTION_STATES = {
+	WAITING_PAYMENT: 'waitingPayment',
+	PENDING: 'pending',
+	WAITING_AUTHORIZATION: 'waitingAuthorization',
+	FAILED: 'failed',
+	COMPLETED: 'completed',
+};
+
+/**
+ * Query params added by MoonPay when redirecting after completing flow
+ * https://www.moonpay.com/dashboard/getting_started/
+ * @typedef MoonPayRedirectTransaction
+ * @type {object}
+ * @property {string} transactionId
+ * @property {string} transactionStatus
+ */
+
+//* Functions
+
+const MOONPAY_ALLOWED_NETWORKS = [
+	NETWORKS_CHAIN_ID.MAINNET,
+	NETWORKS_CHAIN_ID.BSC,
+	NETWORKS_CHAIN_ID.POLYGON,
+	NETWORKS_CHAIN_ID.AVAXCCHAIN,
+	NETWORKS_CHAIN_ID.CELO,
+];
+export const isMoonpayAllowedToBuy = (chainId) => MOONPAY_ALLOWED_NETWORKS.includes(chainId);
+
+const getCurrencyCode = (code) => (code?.indexOf('_') > -1 ? code.split('_')[0] : code)?.toUpperCase();
+
+/**
+ * @param {MoonPayTransaction} transaction
+ */
+const getTransactionFee = (transaction) =>
+	transaction ? transaction.feeAmount + transaction.extraFeeAmount + transaction.networkFeeAmount : 0;
+
+/**
+ * @param {MoonPayTransaction} transaction
+ */
+const getTransactionAmount = (transaction) =>
+	transaction ? transaction.baseCurrencyAmount + getTransactionFee(transaction) : 0;
+
+//* Constants
+
+const { MOONPAY_URL, MOONPAY_URL_STAGING, MOONPAY_API_URL_PRODUCTION, MOONPAY_REDIRECT_URL } = AppConstants.FIAT_ORDERS;
+
+const isDevelopment = process.env.NODE_ENV !== 'production';
+
+const MOONPAY_API_BASE_URL = isDevelopment ? MOONPAY_API_URL_PRODUCTION : MOONPAY_API_URL_PRODUCTION;
```

Comment:
Add a comment to explain this line, which is misleading
```suggestion
// MoonPay uses the same API URL for orders created in their sandbox and production environments
const MOONPAY_API_BASE_URL = isDevelopment ? MOONPAY_API_URL_PRODUCTION : MOONPAY_API_URL_PRODUCTION;
```